client/state/sites/plans/actions.js

```diff
@@ -123,6 +125,21 @@ export function fetchSitePlans( siteId ) {
  */
 export function fetchSitePlansCompleted( siteId, data ) {
 	data = omit( data, '_headers' );
+	const { product_id } = personalPlan;
+
+	if ( ! isEmpty( data ) && ! data.hasOwnProperty( product_id ) ) {
+		const { formatted_price, product_name, product_slug, raw_price } = personalPlan;
+		data[ product_id ] = {
+			can_start_trial: true,
+			discount_reason: null,
+			formatted_discount: '$0',
+			formatted_price,
+			product_name,
+			product_slug,
+			raw_discount: 0,
+			raw_price
+		};
```

Comment:
This is another place to watch out for unexpected mutation. We are directly modifying the passed `data` argument. Now that we have Babel 6 running, it's a great opportunity to take advantage of the spread operator in places like this for immutable alternatives.

```js
const newData = {
	…data,
	[ product_id ]: {
		can_start_trial: true,
		discount_reason: null,
		…
	}
}
```